### PR TITLE
Remove `subject_transform_dest`

### PR DESCRIPTION
**Pull request by jnmoyne**
Removes the single subject transform destination field any subject transformation in StreamSources must now be done using the SubjectTransforms array instead.

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -180,22 +180,24 @@ type PeerInfo struct {
...
-	OptStartSeq          uint64                   `json:"opt_start_seq,omitempty"`
-	OptStartTime         *time.Time               `json:"opt_start_time,omitempty"`
-	FilterSubject        string                   `json:"filter_subject,omitempty"`
+	Name          string     `json:"name"`
+	OptStartSeq   uint64     `json:"opt_start_seq,omitempty"`
+	OptStartTime  *time.Time `json:"opt_start_time,omitempty"`
+	FilterSubject string     `json:"filter_subject,omitempty"`
+	// Do not expose SubjectTransformDest in the client libraries
+	// it is here because it was enabled for a while in the 2.10 beta but is now removed
+	// and there is now a check to error out if it is provided in the request
```
Comment (wallyqs) [inline server/stream.go:200]: If it is an unused field and not in v2.9.22, I think now the best chance to remove it...

Comment (jnmoyne) [reply server/stream.go:200]: That's exactly what this PR is about :)

Comment (wallyqs) [reply server/stream.go:200]: ok we can as is for now but remove in v2.11

Comment (jnmoyne) [reply server/stream.go:200]: If it makes it in 2.10 then has to be maintained afterwards if we want to keep backwards compatibility, hence why right now is the last chance to remove it while it's never been officially part of a 'main' release (even though it was on dev for a very long time)

**Review by wallyqs — APPROVED** [review]
LGTM

**Review by wallyqs — APPROVED** [review]
LGTM

**Review by derekcollison — APPROVED** [review]
LGTM - if tests pass.

**Review by neilalexander — APPROVED** [review]
LGTM now that the `TestJetStreamFilteredSubjectUsesNewConsumerCreateSubject` test exists and passes.

**Pull request merged**